src/inversion_scripts/invert.py

```diff
@@ -297,34 +308,162 @@ def do_inversion(
+        results_dict["hyperparameters"].append(params)
+
+    # Find the ensemble member that is closest to 1 following Lu et al. (2021)
+    idx_best_Ja = np.argmin(np.abs(np.array(results_dict["Ja_normalized"]) - 1))
+    print(
+        f"Best J_A: {results_dict['Ja_normalized'][idx_best_Ja]} with"
+        + f" (prior_err, obs_err, gamma, prior_err_bc, prior_err_oh) = {hyperparam_ensemble[idx_best_Ja]}"
+    )
+
+    # Create an xarray.Dataset
+    dataset = xr.Dataset()
+    for key, values in results_dict.items():
+        if key == "hyperparameters":
+            continue  # Skip hyperparameters; used for attributes
+        for idx, (params, value) in enumerate(
+            zip(results_dict["hyperparameters"], values)
+        ):
+            if isinstance(value, np.ndarray) and value.ndim == 2:
+                dims = ("nvar1", "nvar2")
+            elif isinstance(value, np.ndarray):
+                dims = ("nvar1",)
+            else:
+                dims = ()  # Scalar
+            var_name = f"{key}_{idx + 1}_ensemble_member"
+            dataset[var_name] = (dims, value)
+            dataset[var_name].attrs = params
+
+            # Use the best J_A as the default data variable values
+            if idx == idx_best_Ja:
+                dataset[key] = (dims, value)
+                params_copy = params.copy()
+                params_copy["ensemble_member"] = idx + 1
+                dataset[key].attrs = params_copy
+
+    # reorder the variables, so the default vars are at the top
+    best_vars = list(results_dict.keys())
+    best_vars.remove("hyperparameters")
+    ensemble_vars = [item for item in list(dataset.data_vars) if item not in best_vars]
+
+    new_dataset = xr.Dataset(
+        {var: dataset[var] for var in best_vars + ensemble_vars},
+        coords=dataset.coords,
+    )
+
+    dataset = new_dataset
+
+    return dataset
 
 
 if __name__ == "__main__":
     import sys
 
-    n_elements = int(sys.argv[1])
-    jacobian_dir = sys.argv[2]
-    output_path = sys.argv[3]
-    lon_min = float(sys.argv[4])
-    lon_max = float(sys.argv[5])
-    lat_min = float(sys.argv[6])
-    lat_max = float(sys.argv[7])
-    prior_err = float(sys.argv[8])
-    obs_err = float(sys.argv[9])
-    gamma = float(sys.argv[10])
-    res = sys.argv[11]
-    jacobian_sf = sys.argv[12]
-    prior_err_BC = float(sys.argv[13])
-    prior_err_OH = float(sys.argv[14])
-    is_Regional = sys.argv[15].lower() == "true"
+    config_path = sys.argv[1]
+    n_elements = int(sys.argv[2])
+    jacobian_dir = sys.argv[3]
+    output_path = sys.argv[4]
+    lon_min = float(sys.argv[5])
+    lon_max = float(sys.argv[6])
+    lat_min = float(sys.argv[7])
+    lat_max = float(sys.argv[8])
+    res = sys.argv[9]
+    jacobian_sf = sys.argv[10]
+
+    # read in config file
+    with open(config_path, "r") as f:
+        config = yaml.safe_load(f)
+
+    # set parameters based on config file
+    is_Regional = ensure_float_list(config["isRegional"])
```

Comment:
This won't work for a global inversion: if `config["isRegional"]` is `False`, then `is_Regional` = `[0.0]`, and the rest of the script proceeds as if it is a regional inversion. 
Maybe set `is_Regional = config["isRegional"]` so that the boolean gets passed into do_inversion_ensemble?

Reply:
Good catch, fixed!